### ISSUE TITLE
[Bug, analyzed] StartCounter.txt can unrightfully get stuck at "3" forever, preventing Startup

### ISSUE THREAD

**Issue by wolframs** [microsoft/navcontainerhelper#3222]
**Issue in short:**
If the _NavStart.ps1 / Start.ps1 procedure_ during Docker Container start gets _ungracefully_ interrupted, the StartCounter.txt content can get stuck at "3".
That then prevents any succesful startup of the container, even though no actual error occurs.

**If you're here from Google: Workardound:**
The following command resets the Start Counter, avoiding the issue I describe:
`Invoke-ScriptInBcContainer -containerName $containerName -scriptblock { Set-Content -path 'c:\run\startcount.txt' -Value '0' }`

**How to Reproduce the bug:**
Repeatedly (!) start the docker container and ungracefully kill the ensuing processes. (E.g. kill the docker engine process during container initialization, etc.)

**Probable cause:**
The StartCounter can get incremented during every start up sequence.
But it is never reset, if the sequence gets inappropriately aborted.

**Why is this even an issue?**
This happened to me while creating an automation script, to run certain tasks, including the docker engine startup and container startup. It has happened to colleagues of mine, when they were troubleshooting issues, that they deemed related to the NAV Service Tier, and when they weren't aware of the possibility to restart the Service Tier without restarting the docker container.

**List of involved files, Paths in the Container:**
```
"C:\Run\Start.ps1"
"C:\Run\NavStart.ps1"
"C:\Run\StartCounter.txt"
```

**Output of troubled startup:**
```
Initializing...
Setting host.docker.internal to 192.168.178.63 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 192.168.178.63 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.25.64.1 in container hosts file
Restarting Container
PublicDnsName unchanged
Restart count 3
Error starting container
at <ScriptBlock>, C:\Run\navstart.ps1: line 50
at <ScriptBlock>, C:\Run\start.ps1: line 384
at <ScriptBlock>, <No file>: line 1
```

**Screenshots**
![image](https://github.com/microsoft/navcontainerhelper/assets/5008890/f1a4566e-caf2-40d5-862d-cc34a15f6763)

**Comment by freddydk** [comment]
Well, this is the purpose of the counter - if the container has 3 consecutive attempts to start which were unsuccessful - then it will not try again.
This is to prevent containers from running into a never ending restart loop where they just eat up all resources on the host without ever doing anything valuable.

The counter should be reset when the container successfully starts.

**Comment by wolframs** [comment]
Hi @freddydk, that exactly is the bug: The counter does _not_ reset in some rare cases ;)
That happens when the container start is _attempted, but interrupted_ repeatedly (at least 3 times, apparently). The counter then increments but is never reset on each attempt.

The concrete issue here is, that **the start will never even be attempted anymore** if the content of "StartCounter.txt" is 3.

This could be remedied by either a command like Reset-BCContainerStartCounter or by arranging the NavStart.ps1 structure such that the start is always attempted at least once, no matter the contents of StartCounter.txt.

**Comment by freddydk** [comment]
I am totally aware of the behavior, but the reason for putting in the counter was, that on certain versions of windows, the container went into an infinite restart loop (restarting automatically during initialization) and ate resources from the host computer to a point, where people would think there was something seriously wrong with their computer.
This is why I chose to say that after 3 unsuccessful attempts to start the container - why should we try a 4th time?
Now you are stating that if the user interrupted the container from being started 3 times - it will be seen as being a never-ending restart loop - that is correct, but the user can just run this line:

```
Invoke-ScriptInBcContainer -containerName $containerName -scriptblock { Set-Content -path 'c:\run\startcount.txt' -Value '0' }
```

and then restart the container again - that should work.
I could do one thing... - in Restart-BcContainer, I could add the above line before restarting (if the container is running) - this means that Restart-BcContainer would always retry 3 times again - I guess this would be a good fix, or?

**Comment by freddydk** [comment]
Shipped in 6.0.5
